sound/soc/sof/intel/mtl.c

```diff
@@ -612,6 +613,27 @@ static int mtl_dsp_disable_interrupts(struct snd_sof_dev *sdev)
 	return mtl_disable_interrupts(sdev);
 }
 
+static int mtl_dsp_get_stream_position(struct snd_sof_dev *sdev,
+				       struct snd_soc_component *component,
+				       struct snd_pcm_substream *substream)
+{
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct hdac_stream *hstream = substream->runtime->private_data;
+	struct snd_soc_component *scomp = sdev->component;
+	struct sof_ipc_stream_posn *posn;
+	struct snd_sof_pcm *spcm;
+	u32 llp_l, llp_u;
+
+	spcm = snd_sof_find_spcm_dai(scomp, rtd);
+	posn = &spcm->stream[substream->stream].posn;
+
+	llp_l = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, MTL_PPLCLLPL(hstream->index));
+	llp_u = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, MTL_PPLCLLPU(hstream->index));
+	posn->dai_posn = ((u64)llp_l | (u64)llp_u << 32);
```

Comment:
Did @ujfalusi you intend to have a u64 return type and "return ((u64)llp_l | (u64)llp_u << 32);" ..?

Reply:
change to return u64